c++ compiler output/errors not html escaped

**What Course are you in**
thinkcpp

**What Page were you on**
any with active code/c++

**What is your username**

**Describe the bug**
When compiling code with an include error, like:

```
#include <iostreeeeeem>
```
The error window appears to not properly escape the <'s and >'s.

```
test.cpp:1:10: fatal error: iostreeeeeem: No such file or directory
    1 | #include 
      |          ^~~~~~~~~~~~~~
compilation terminated.
```

Similarly, if I include code like:

```
cout << "Hello, <World>." << endl;
```

The output looks like:

```
Hello, .
```

I'm not sure how the runestone pieces all fit together, but I think the bug is likely in livecode.js. It appears to have handlers to substitute &lt;br&gt; for \n, but I think it needs to do a full htmlification (is that a word?) on the stdout of the program and also on the compiler errors/warnings.

--- a/bases/rsptx/interactives/runestone/activecode/js/livecode.js
+++ b/bases/rsptx/interactives/runestone/activecode/js/livecode.js
@@ -333,7 +333,7 @@ export default class LiveCode extends ActiveCode {
                     result.stdout,
                     this.divid
                 );
-                $(odiv).html(this.parsedOutput.stdout);
+                $(odiv).html(escapeHtml(this.parsedOutput.stdout));
                 if (this.suffix) {
                     if (this.parsedOutput.pct === undefined) {
                         this.parsedOutput.pct =
@@ -356,7 +356,7 @@ export default class LiveCode extends ActiveCode {
                 }
                 break;
             case 13: // time limit
-                $(odiv).html(result.stdout.replace(/\n/g, "<br>"));
+                $(odiv).html(escapeHtml(result.stdout.replace(/\n/g, "<br>")));
                 this.addJobeErrorMessage(
                     $.i18n("msg_activecode_time_limit_exc")
                 );

--- a/bases/rsptx/interactives/runestone/activecode/js/livecode.js
+++ b/bases/rsptx/interactives/runestone/activecode/js/livecode.js
@@ -397,7 +397,7 @@ export default class LiveCode extends ActiveCode {
         eContainer.id = this.divid + "_errinfo";
         eContainer.appendChild(errHead[0]);
         var errText = eContainer.appendChild(document.createElement("pre"));
-        errText.innerHTML = err;
+        errText.innerHTML = escapeHtml(err);
     }
     /**
      * Checks to see if file is on server
@@ -659,5 +659,15 @@ function unescapeHtml(safe) {
             .replace(/&gt;/g, ">")
             .replace(/&quot;/g, '"')
             .replace(/&#x27;/g, "'");
+    }
+}
+function escapeHtml(str) {
+    if (str) {
+	return str
+	    .replace(/&/g, '&amp;')
+	    .replace(/</g, '&lt;')
+	    .replace(/>/g, '&gt;')
+	    .replace(/'/g, '&#x27;')
+	    .replace(/"/g, '&quot;');
     }
 }

